packages/beachball/src/bump.ts

```diff
@@ -79,11 +80,48 @@ export function performBump(
     fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2) + '\n');
   });
 
+  // If --bump-deps is set, update all dependent package.json's
+  if (bumpDeps) {
+    const bumpedPackages = Object.keys(packageChangeTypes);
+    const bumpedDependents: string[] = [];
+    let bumpedFlag = false;
+
+    do {
+      bumpedFlag = false;
+      Object.keys(packageInfos).forEach(pkgName => {
+        if (bumpedPackages.includes(pkgName)) {
+          return;
+        }
+
+        const info = packageInfos[pkgName];
+        const packageJsonPath = info.packageJsonPath;
+        const packageJson = JSON.parse(fs.readFileSync(packageJsonPath).toString());
+
+        const allDeps = { ...packageJson.dependencies, ...packageJson.devDependencies };
+        for (const dep of Object.keys(allDeps)) {
+          if (bumpedPackages.includes(dep)) {
+            packageJson.version = semver.inc(packageJson.version, "patch");
```

Comment:
Therein lies the question: do we bump a patch for the indirect bumps? What if the bumped version was a minor, do we bump patch in this? Also, what if in a single patch bump, the dep's external dep got bumped a major version - what do we do then?

There's a lot of strategies here - this is a good start though. I HOPE this is good enough for 80% case.

Reply:
Good catch.

I've thought about this, and I think the best approach will be to **not** use `--bump-deps` flag, but run `beachball change; beachball bump` multiple times and manually choose each package's bump type. We'll test this flag in one or two of our internal repos and see how things go.

Also updated PR desc mentioning this point.